lib/core/context.c

```diff
@@ -190,3 +190,69 @@ void h2o_context_update_timestamp_string_cache(h2o_context_t *ctx)
     h2o_time2str_rfc1123(ctx->_timestamp_cache.value->rfc1123, &gmt);
     h2o_time2str_log(ctx->_timestamp_cache.value->log, ctx->_timestamp_cache.tv_at.tv_sec);
 }
+
+void h2o_context_close_idle_connections(h2o_context_t *ctx, size_t max_connections_to_close, uint64_t min_age)
+{
+    if (max_connections_to_close <= 0)
+        return;
+
+    size_t closed = ctx->_conns.num_conns.shutdown;
+
+    if (closed >= max_connections_to_close)
+        return;
+
+    H2O_CONN_LIST_FOREACH(h2o_conn_t * conn, ({&ctx->_conns.idle}), {
+        struct timeval now = h2o_gettimeofday(ctx->loop);
+        if (h2o_timeval_subtract(&conn->connected_at, &now) < (min_age * 1000))
+            continue;
+        ctx->connection_stats.idle_closed++;
+        conn->callbacks->close_idle_connection(conn);
+        closed++;
+        if (closed == max_connections_to_close)
+            return;
+    });
+}
+
+static size_t *get_connection_state_counter(h2o_context_t *ctx, h2o_conn_state_t state)
+{
+    return ctx->_conns.num_conns.counters + (size_t)state;
+}
+
+static void _connection_state_set(h2o_conn_t *conn, h2o_conn_state_t state)
```

Comment:
```suggestion
static void set_conn_state(h2o_conn_t *conn, h2o_conn_state_t state)
```
Maybe? Some rationale:
* this is function is internal by its linkage being static, we do not need `_`.
* when naming a static function, it is our convention to start with a verb, omitting the name of the module.